pages/spec/naming.md

```diff
@@ -0,0 +1,171 @@
+# Module name requirements
+
+:::{note}
+Module naming requirements only apply to packages which are uploaded to a fpm registry; by default, no naming rules are enforced for local fpm projects.
+:::
+
+The Fortran language does not support namespaces. This means that all public names (modules, but also global subroutines and functions) must be unique in the build space.
+Any build which contains duplicate names will fail, because it is impossible to resolve a name to a unique object.
+For this reason, the fpm registry requires all registered packages to comply with some simple naming conventions, that should apply to both the package name and all its modules.
+
+## Fortran names: general rules
+
+As of Fortran 2003 onward, valid Fortran names need to comply with the following rules:
+
+- Up to 63 characters long;
+- Letters are case insensitive;
+- Must begin with a letter;
+- Only alphanumeric characters (letters, numbers) and underscores `_` are allowed.
+
+*Examples of invalid Fortran names:*
+
+```{code-block} fortran
+  1_package         ! Begins with #
+  package$          ! Contains invalid symbol
+  _package          ! Does not begin with letter
+  my package        ! Contains space
+```
+
+*Examples of valid Fortran names:*
+
+```{code-block} fortran
+  my_module         ! Case insensitive: all versions valid,
+  My_Module         ! but resolving to the same object
+  MY_MODULE
+  MyModule
+  mypackage
+  package_module    ! Underscores allowed
+  my_package_123
+```
+
+## FPM registry names: rules for packages and modules
+
+In order to reduce the chance of name collisions, ``all Fortran modules`` in all packages of a registry are required to begin with their package name.
+The following rules apply:
+
+### Module names
+
+1. Must be valid Fortran names (see above).
+2. Must begin with their package name
+3. Having the exact same name as the package is allowed;
+4. If longer, the ``default separator`` `__` between the package name chunk and what follows must be used.
```

Comment:
What does "longer" mean?

Reply:
That is not just the package name. Agreed: ugly sentence, I removed it